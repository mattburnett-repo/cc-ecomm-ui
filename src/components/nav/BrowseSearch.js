export default function BrowseSearch () {
    return (
        <div role="presentation" aria-label="browse-search">
            <div role="presentation" aria-label="product-categories">select product categories</div>
            <input aria-label="search-terms" placeholder="Enter search terms"></input>
            <button aria-label="search-button">Search</button>
        </div>
    )           
}