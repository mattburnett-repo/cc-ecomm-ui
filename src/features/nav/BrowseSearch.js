import { useSelector } from "react-redux";

import CheckLoginStatus from "../../utils/CheckLoginStatus";
import BrowseSearchDisplay from '../../components/nav/BrowseSearchDisplay';

import { selectProductCategories } from "../productCategory/productCategorySlice";
// eslint-disable-next-line 
import { getProducts, getProductsByCategory } from "../products/productsSlice";

export default function BrowseSearch () {
    CheckLoginStatus()

    // TODO: need to figure out how to get dispatch() to work from here
    function handleChange(e){
        let categoryId = e.target.value

        // eslint-disable-next-line
        if(categoryId == 0) {
            alert('browseSearch select all products. \nneed to figure out how to get dispatch() to work from here')
        } else {
            alert('browseSearch select products for category: ' + categoryId + '. \nneed to figure out how to get dispatch() to work from here')
        }
    }

    function handleSearch(e){
        e.preventDefault()
        // let searchTerms = e.target.value
        alert('handleSearch for search term/s ' + e.target.searchTerms.value + '. \nneed to figure out how to get dispatch() to work from here')
    }
        
    function handlers() {
        return {
            handleChange: handleChange,
            handleSearch: handleSearch,
        }
    }

    const productCategories = useSelector(selectProductCategories)

    return <BrowseSearchDisplay  data={ productCategories } handlers={handlers()}/>
}