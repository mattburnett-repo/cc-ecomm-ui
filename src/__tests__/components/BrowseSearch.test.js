// browse and search functionality common to several other components

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import BrowseSearch from '../../components/nav/BrowseSearch';

const mockProductCategories = [
    {
        "category_id": 4,
        "description": "Auto"
    },
    {
        "category_id": 3,
        "description": "Health"
    },
    {
        "category_id": 1,
        "description": "Pets"
    },
    {
        "category_id": 2,
        "description": "Tech"
    }
];

describe('<BrowseSearch /> component tests', () => {
    beforeEach(() => render(<BrowseSearch />));

    it('should show product categories', () => {
        screen.getByRole('presentation', {name: /product-categories/i});
    });
    it('should show search term input', () => {
        screen.getByRole('textbox', {name: /search-terms/i});
    });
    it('should show search button', () => {
        screen.getByRole('button', {name: /search/i});
    });
}); // end component

describe('<BrowseSearch /> feature tests', () => {
    beforeEach(() => render(<BrowseSearch />));

    it('displays product categories', () => {
        expect(true).toBeFalsy();
    })

    it('creates a snapshot', () => {
        const tree = renderer.create(<BrowseSearch />).toJSON();
        expect(tree).toMatchSnapshot();  
    })
}); // end feature

describe('<BrowseSearch /> unit / functional tests', () => {
    beforeEach(() => render(<BrowseSearch />));

    it('selects a product category', () => {
        expect(true).toBeFalsy();
    })

    it('accepts search text', () => {
        const theElement = screen.getByRole('textbox', {name: /search-terms/i});
        userEvent.type(theElement, 'here is some search text');
        
        expect(theElement).toHaveValue('here is some search text');
    });
    it('clicks the search button', () => {
        const theVal = screen.getByRole('button', {name: /search/i});
        fireEvent.click(theVal);
    });
}); // end feature