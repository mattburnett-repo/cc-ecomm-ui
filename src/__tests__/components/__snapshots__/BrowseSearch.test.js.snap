// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowseSearch /> feature tests creates a snapshot 1`] = `
<div
  aria-label="browse-search"
  role="presentation"
>
  <div
    aria-label="product-categories"
    role="presentation"
  >
    select product categories
  </div>
  <input
    aria-label="search-terms"
    placeholder="Enter search terms"
  />
  <button
    aria-label="search-button"
  >
    Search
  </button>
</div>
`;
